command/agent/alicloud_end_to_end_test.go

```diff
@@ -100,13 +100,9 @@ func TestAliCloudEndToEnd(t *testing.T) {
 		Client: client,
 	}
 
-	errCh := make(chan error)
 	ah := auth.NewAuthHandler(ahConfig)
-	go ah.Run(ctx, am, errCh)
-	defer func() {
-		select {
-		case <-ah.DoneCh:
-		case err := <-errCh:
+	go func() {
```

Comment:
IIRC you can't t.Fatal() in a go routine, maybe for tests you can send the error down an errCh to avoid this?

Reply:
:( Ah right, I forgot..

Reply:
You could also use t.Run() with t.Parallel(), kinda hacky though.

Reply:
Ran with Brian's suggestion, should be addressed now.